src/test/java/org/prebid/server/bidder/grid/GridBidderTest.java

```diff
@@ -83,9 +94,93 @@ public void makeHttpRequestsShouldReturnErrorIfExtImpGridNull() {
         final Result<List<HttpRequest<BidRequest>>> result = gridBidder.makeHttpRequests(bidRequest);
 
         // then
-        assertThat(result.getErrors()).hasSize(1);
-        assertThat(result.getErrors().get(0).getMessage()).startsWith("uid is empty");
-        assertThat(result.getErrors().get(0).getType()).isEqualTo(BidderError.Type.bad_input);
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue())
+                .extracting(HttpRequest::getPayload)
+                .flatExtracting(BidRequest::getImp)
+                .extracting(Imp::getExt)
+                .containsExactly(
+                        mapper.valueToTree(ExtImpGrid.builder()
+                                .data(ExtImpGridData.of("pbadslot",
+                                        ExtImpGridDataAdServer.of("name", "adslot")))
+                                .bidder(ExtImpGridBidder.of(1, null))
+                                .gpid("adslot")
+                                .build()));
+    }
+
+    @Test
+    public void makeHttpRequestsShouldCorrectlyModifyRequestExt() throws JsonProcessingException {
+        // given
+        final ObjectNode impUserKeywordsNode = (ObjectNode) mapper.readTree(
+                "{\"firstPublisher\":[{\"name\":\"firstKeywordsUserSection\","
+                        + "\"segments\":[{\"name\":\"segment1\",\"value\":\"value1\"}]}]}");
+        final ObjectNode impSiteKeywordsNode = (ObjectNode) mapper.readTree(
+                "{\"firstPublisher\":[{\"name\":\"firstKeywordsSiteSection\","
+                        + "\"segments\":[{\"name\":\"segment1\",\"value\":\"value1\"}]}]}");
+        final ExtImpGrid impExt = ExtImpGrid.builder()
+                .data(ExtImpGridData.of("pbadslot",
+                        ExtImpGridDataAdServer.of("name", "adslot")))
+                .bidder(ExtImpGridBidder.of(1, Keywords.of(impUserKeywordsNode, impSiteKeywordsNode)))
+                .build();
+
+        final ExtRequest extRequest = ExtRequest.of(null);
+        extRequest.addProperty("keywords", mapper.readTree(
+                "{\"user\": {\"secondPublisher\":[{\"name\":\"secondKeywordsUserSection\","
```

Comment:
If we will add something like `"stringKey": "stringVal",` before user, it will not be in result ext, but it is in go